glambie/processing/processing_helpers.py

```diff
@@ -56,6 +63,19 @@ def filter_catalogue_with_config_settings(data_group: GlambieDataGroup,
             datasets_annual = [d for d in datasets_annual if d.user_group.lower() != ds.lower()]
     if data_group == GLAMBIE_DATA_GROUPS["demdiff_and_glaciological"]:
         datasets_annual = [d for d in datasets_annual if d.data_group != GLAMBIE_DATA_GROUPS["demdiff"]]
+    # 2.1 add combined datasets
+    additional_annual_datasets = []
+    if "include_combined_annual_datasets" in region_config.region_run_settings[data_group.name]:
+        include_combined_annual_datasets = region_config.region_run_settings[data_group.name].get(
+            "include_combined_annual_datasets", [])
+        include_combined_annual_datasets = [x for x in include_combined_annual_datasets if x is not None]
+        for ds in include_combined_annual_datasets:
+            catalogue_filtered = data_catalogue_original.get_filtered_catalogue(
+                data_group="combined", region_name=region_config.region_name, user_group=ds)
+            if len(catalogue_filtered.datasets) > 0:
+                additional_annual_datasets.append(catalogue_filtered.datasets[0])
+            else:
+                log.info("Cannot find and add the following combined dataset to annual datasets: %s", ds)
```

Comment:
This and rows 90->101 are duplicates. Abstract to a simple, private function and call for annual datasets and trends. This will simplify the readability of this function and make testing simpler.

Reply:
Fair! Created a function called get_additional_combined_datasets()